src/main/java/Duke.java

```diff
@@ -1,10 +1,222 @@
+                    newTask = new Deadline(description, by);
+                }
+
+                TASKS.add(newTask);
+                //checks whether this tasks is marked as done or not
+                if (pointer.substring(3).startsWith("[X]")) {
+                    newTask.mark();
+                }
+            }
+            reader.close();
+        }
+    }
+
+    private static void writeFile() {
+        Scanner reader = null;
+        String text = "";
+        try {
+            Files.deleteIfExists(tempTasksFilePath);
+            Files.createFile(tempTasksFilePath);
+            for (int i = 0; i < TASKS.size(); i++) {
+                text += TASKS.get(i) + "\n";
+            }
+            Files.writeString(tempTasksFilePath, text);
+            Files.deleteIfExists(tasksFilePath);
+            Files.move(tempTasksFilePath, tasksFilePath);
+        } catch (IOException e) {
+            System.out.print(e.getMessage());
+        } finally {
+            if (reader != null) {
+                reader.close();
+            }
+        }
+    }
+
+    private static void addTask(Task newTask) {
+        TASKS.add(newTask);
+        writeFile();
+        System.out.println(reply("Got it. I've added this task:\n  "
+                    + newTask
+                    + "\nNow you have " + TASKS.size() + " tasks in the list\n"));
+
+    }
+
+    private static void mark(String command) {
+        int index = Integer.valueOf(command.substring(5)) - 1;
+        Task target = TASKS.get(index);
+        target.mark();
+        writeFile();
+        System.out.println(reply("Nice! I've marked this task as above:\n  " + target + "\n"));
+    }
+
+    private static void unmark(String command) {
+        int index = Integer.valueOf(command.substring(7)) - 1;
+        Task target = TASKS.get(index);
+        target.unmark();
+        writeFile();
+        System.out.println(reply("OK, I've marked this task as not done yet:\n  " + target + "\n"));
+    }
+
+    private static void displayTasks() {
+        String list_of_tasks = "";
+        Task currentTask;
+        for (int i = 0; i < TASKS.size(); i++) {
+            currentTask = TASKS.get(i);
+            list_of_tasks += i + 1 + "." + currentTask.toString() + "\n";
+        }
+        System.out.println(reply("Here are the tasks in your list:\n" + list_of_tasks));
+    }
+
+    private static void delete(String command) {
+        int index = Integer.valueOf(command.substring(7)) - 1;
+        Task target = TASKS.get(index);
+        System.out.println(reply("Noted. I've removed this task:\n"
+                + target + "\nNow you have " + TASKS.size() + "in the list\n"));
+        TASKS.remove(index);
+        writeFile();
+    }
+
+    private static void handleTaskInput(Scanner input) {
+        String command = input.nextLine();
+        while (true) {
+            if (command.equals("bye")) {
+                break;
```

Comment:
Just a suggestion but method seems to be quite long. 
Perhaps breaking it up into smaller methods , by giving a method for each command will make it easier to maintain 
and debug.